application/controllers/survey_response.py

```diff
@@ -112,32 +121,35 @@ def _encrypt_message(message_json):
         encrypter = Encrypter(json_secret_keys)
         return encrypter.encrypt(message_json)
 
-    def generate_file_name(self, case_id, file_extension):
+    def get_file_name_and_survey_ref(self, case_id, file_extension):
```

Comment:
I think combining these two bit of functionality in one function here isn't ideal, returning 2 values will making testing a bit more convoluted.
The actual creation of the file name should probably be separate from the calls to get the info for it.
This is probably more to do with the current structure of the code rather than this PR

Reply:
You are right , it rubbed me up the wrong way . But when you look at the code it goes to case_group, collection_exercise and then survey which is 3 service calls . And its difficult to extract either a survey_ref or filename without doing the same expensive steps . So I added a comment as to why I did this, personally in this case I would leave as is . If I knew that the services all used some sort of caching then I would say split them . This might be an argument for adding caching to all services . It does not have to be complex , it could be a simple dictionary with a request lifespan . If we had that then calling these services for the same information would involve 1 http call and one dictionary lookup. Then I would agree , split them . It might be worth raising that . Its a trivial thing to add and could lead to better looking code ? I might pick that up with Mark , or feel free to raise it ?

Reply:
Functionally it all seems to be working fine so I'm happy to leave as is.

I think a solution could be to make all the required external calls in the view sequentially then have a generate_file_name function which takes all the required elements (exercise_ref/survey_ref) directly

Reply:
It would be good to understand how these python 'classes' are handled re: scope. Having an equivalent of a request scoped spring 'bean' would probably give us some options. We will be looking at reddis for caching at some point, which I imagine will initially cache (the relatively static) survey and CE meta data that we are  concerned with here.